gstd/src/async_runtime/locks.rs

```diff
@@ -164,51 +183,152 @@ impl LocksMap {
         //
         // Locks with `deadline <= now`, they will be removed in the following polling.
         let now = exec::block_height();
-        map.iter()
-            .filter_map(|(_, lock)| (lock.deadline() > now).then_some(lock))
+
+        crate::log!(
+            "wait({message_id:.2?}): current block is {}",
+            crate::util::u32_with_sep(now)
+        );
+
+        crate::log!("wait({message_id:.2?}): looking for appropriate lock to be waited");
+
+        if let Some(lock) = map
+            .iter()
+            .filter_map(|(_, lock)| {
+                if lock.deadline() > now {
+                    crate::log!("wait({message_id:.2?}): lookup... accepting {lock:.2?}");
+
+                    Some(lock)
+                } else {
+                    crate::log!("wait({message_id:.2?}): lookup... filtering out {lock:.2?}");
+
+                    None
+                }
+            })
             .min_by(|lock1, lock2| lock1.cmp(lock2))
-            .expect("Cannot find lock to be waited")
-            .wait();
+        {
+            crate::log!("wait({message_id:.2?}): waiting closest {lock:.2?}");
+
+            lock.wait();
+        } else {
+            crate::log!("wait({message_id:.2?}): couldn't find appropriate lock, panicking...");
+
+            panic!("Couldn't find lock to be waited");
+        }
     }
 
     /// Lock message.
     pub fn lock(&mut self, message_id: MessageId, waiting_reply_to: MessageId, lock: Lock) {
-        let locks = self.0.entry(message_id).or_insert_with(Default::default);
-        locks.insert(LockContext::ReplyTo(waiting_reply_to), lock);
+        crate::log!("lock({message_id:.2?}, {waiting_reply_to:.2?}, {lock:.2?})");
```

Comment:
I think this message is too indistinguishable, i.e. it will be difficult to find source code where this log message was emitted by the log message content